gpflow/conditionals.py

```diff
@@ -36,55 +35,91 @@
+    Single-output GP allowing repetitions
+    :param f: M x R
+    :param q_sqrt: M x R  or  R x M x M
+    :return: N x R  or R x N x N  or  N x R x R  or  N x R x N x R
+    """
+    Kmm = feat.Kuu(kern, jitter=settings.numerics.jitter_level)  # M x M
+    Kmn = feat.Kuf(kern, Xnew)  # M x N
     if full_cov:
-        Knn = kern.K(Xnew)
+        Knn = kern.K(Xnew)  # N x N
     else:
-        Knn = kern.Kdiag(Xnew)
-    return base_conditional(Kmn, Kmm, Knn, f, full_cov=full_cov, q_sqrt=q_sqrt, white=white)  # N x R,  N x R
+        Knn = kern.Kdiag(Xnew)  # N
+    fmean, fvar = base_conditional(Kmn, Kmm, Knn, f, full_cov=full_cov, q_sqrt=q_sqrt, white=white)  # N x R,  N x R
+    return fmean, expand_independent_outputs(fvar, full_cov, full_cov_output)
 
 
 @dispatch(IndependentFeature, IndependentMultiKernel, object, object)
 @name_scope()
 def feature_conditional(feat, kern, Xnew, f, *, full_cov=False, full_cov_output=False, q_sqrt=None, white=False):
     """
-    Multi-output GP with independent GP priors
-    :param Xnew:
-    :param feat:
-    :param kern:
-    :param f: M x L x R  or  M x L x 1
-    :param full_cov:
-    :param full_cov_output:
-    :param q_sqrt: R x P x M  or R x P x M x M
-    :param white:
-    :return:
+    Multi-output GP with independent GP priors.
+    Number of latent processes equals the number of outputs (L = P). Expected kernels:
+     Kmm
+    :param f: M x P
+    :param q_sqrt: M x P  or  P x M x M
+    :return: N x P ,
     """
-    if f.shape.ndims != 3 or q_sqrt.shape.ndims != 4:
-        raise GPflowError("IndependentFeature & IndependentMultiKernel combination requires separated GP posterior "
-                          "representations, i.e. f: R x M x P.")
-
     # Following are: P x M x M  -  P x M x N  -  P x N(x N)
     Kmms = feat.Kuu(kern, jitter=settings.jitter)
     Kmns = feat.Kuf(kern, Xnew)
+    # TODO: Do all independent kernels have a kern_list?
     Knns = tf.stack([kern.K(Xnew) if full_cov else kern.Kdiag(Xnew) for kern in kern.kern_list], axis=0)
-    fs = tf.transpose(f, [1, 0, 2])  # P x M x R
-    q_sqrts = tf.transpose(q_sqrt, [1, 0, 2, 3])
+    fs = tf.transpose(f)[:, :, None]  # P x M x 1
+    # P x 1 x M x M  or  P x M x 1
+    q_sqrts = tf.transpose(q_sqrt)[:, :, None] if q_sqrt.shape.ndims == 2 else q_sqrt[:, None, :, :]
 
     def single_gp_conditional(t):
         Kmm, Kmn, Knn, f, q_sqrt = t
         return base_conditional(Kmn, Kmm, Knn, f, full_cov=full_cov, q_sqrt=q_sqrt, white=white)
 
     rmu, rvar = tf.map_fn(single_gp_conditional,
                           (Kmms, Kmns, Knns, fs, q_sqrts),
-                          (settings.float_type, settings.float_type))  # P x N x R  ,  P x N(x N)x R
+                          (settings.float_type, settings.float_type))  # P x N x 1  ,  P x N(x N) x 1
 
-    rmu = tf.transpose(rmu[:, :, 0])
-    rvar = tf.transpose(rvar[:, :, 0])
-    return rmu, rvar
+    if full_cov or full_cov_output:
+        raise NotImplementedError
```

Comment:
Why aren't we implementing this? 
We would like to use this case but with `full_cov` and/or `full_cov_output` equal to true.

Reply:
Just haven't got round to it. I stared a convenience function up top to do make the matrix diagonals etc (in this case all these output covariances are diagonal).